src/src_cryptography/src_libgcrypt/cryptography_interface_libgcrypt.template.c

```diff
@@ -823,6 +833,142 @@ static int32_t cryptography_validate_authentication(uint8_t* data_out, size_t le
     gcry_mac_close(tmp_mac_hd);
     return status; 
 }
+
+static int32_t cryptography_encrypt(uint8_t* data_out, size_t len_data_out,
+                                         uint8_t* data_in, size_t len_data_in,
+                                         uint8_t* key, uint32_t len_key,
+                                         SecurityAssociation_t* sa_ptr,
+                                         uint8_t* iv, uint32_t iv_len,uint8_t* ecs, uint8_t padding)
+{
+    gcry_error_t gcry_error = GPG_ERR_NO_ERROR;
+    gcry_cipher_hd_t tmp_hd;
+    int32_t status = CRYPTO_LIB_SUCCESS;
+    uint8_t* key_ptr = key;
+
+    data_out = data_out;        // TODO:  Look into tailoring these out, as they're not used or needed.
+    len_data_out = len_data_out;
+    padding = padding;
+
+    if(*ecs == CRYPTO_CIPHER_AES256_CBC && crypto_config->padding_control == KMC_PADDING)
+    {
+        // Nothing necessary Here for now
```

Comment:
Delete the unnecessary code block.

Reply:
This line is also solving the same issue lines 843 through 846 are.  It's just a placeholder for later if it is needed, and replaces ecs=ecs.  Would you rather it be ecs=ecs?

Reply:
Are you saying this PR does not implement the libgcrypt version of CBC AES25 encrypt? You'll never need KMC_PADDING or LGC_PADDING controls -- you should know that already by the fact that the function is called in that template file.

Reply:
And yes, to handle the unused variable checks that error out, it's ok to use the self setting trick...

Reply:
No, that's not what I'm saying at all.  That piece is just a place holder.
The padding controls are necessary.  KMC will not accept a pre-padded frame.  LibGcrypt requires us to pad the frame. So the bytes need to be removed.  It might be possible to pass it in differently than with the config, though, but we'll still need some kind of tracking / control when passing to either the libgcrypt.template, or the kmc.template.

If I've made anymore sense?

Reply:
You're explaining why these controls are not necessary in your own statement above... The padding_control is a transient property that can be determined automatically if the user is using libgcrypt vs the kmc crypto service cryptography interface... Users do need to set this at all! If they use libgcrypt, you pass in the pad frame, if they use kmc, you don't... That's it -- no special configs required.

Reply:
Done

Reply:
Confirmed removed in commit 7b13732